sdks/python/apache_beam/io/restriction_trackers.py

```diff
@@ -92,7 +92,8 @@ def __init__(self, offset_range):
     self._checkpointed = False
 
   def check_done(self):
-    if self._last_claim_attempt < self._range.stop - 1:
+    if (self._range.start != self._range.stop and
+        self._last_claim_attempt < self._range.stop - 1):
```

Comment:
`self._last_claim_attempt` could be `None`.  You may want to have `self._range.start != self._range.stop and (self._last_claim_attemp is None or self._last_claim_attempt < self._range.stop - 1)`

Reply:
Good point

Reply:
Fixed